hangman.py

```diff
@@ -0,0 +1,191 @@
+        if letter not in old_letters_guessed:
+            hidden_word += "_ "
+        else:
+            hidden_word += letter + " "
+    return hidden_word[:-1]
+
+def try_update_letter_guessed(letter_guessed, old_letters_guessed):
+    """
+    the funcs receive the user's guess and old guesses and  add his guess to the list of guesses if the letter is one alphabetic letter.
+    :param letter_guessed - the letter the user gueesed, string:
+    :param old_letters_guessed - the guess he gussed before, list:
+    :return if the letter is ok, boolean:
+    """
+    if check_valid_input(letter_guessed, old_letters_guessed):
+        return True
+    else:
+        print("X - not a valid input, enter 1 new alphabetic letter")
+        if old_letters_guessed:
+            print("the letters you already guessed are:", " -> ".join(sorted(old_letters_guessed)))
+        return False
+
+def check_valid_input(letter_guessed, old_letters_guessed):
+    """
+    the func tells if the letter guessed is only one alphabetic letter and was not guessed before
+    :param letter_guessed - the letter we check, string:
+    :param old_letter_guessed - all the letters the player already guessed, list:
+    :return if the letter is valid and was not guessed before, boolean:
+    """
+    letter_guessed = letter_guessed.lower()
+    return len(letter_guessed) == 1 and letter_guessed.isalpha() and letter_guessed not in old_letters_guessed
+
+def check_win(secret_word, old_letters_guessed):
+    """
+    the funcs checks if the player won and guessed all the letters
+    :param secret_word - the word he needs to guess, a string:
+    :param old_letters_guessed - the letters he had already guessed, a list:
+    :return - void:
+    """
+    return "_" not in show_hidden_word(secret_word, old_letters_guessed)
+
+def guessing_a_letter(old_letters_guessed, secret_word, num_of_tries):
+    """
+    the funcs lets the player guess a letter, show him the state of the word and the state of the hangman and old
+    guessed letters if he guessed wrong
+    :param old_letters_guessed - all the letters he already guessed - list:
+    :param secret_word - the word he needs to guess - string:
+    :param num_of_tries - the number of wrong guesses - num:
+    :return - the updated list of the letters he guessed - list, the updated number of wrong guesses - num:
+    """
+    letter_guessed = input("Guess a letter:").lower()
+    while not try_update_letter_guessed(letter_guessed, old_letters_guessed): #i want to check if the letter is valid
+        letter_guessed = input("Guess a letter:").lower()
+    old_letters_guessed.append(letter_guessed)
+    if letter_guessed not in secret_word:
+        print(":(")
+        num_of_tries += 1
+        print_hangman(num_of_tries)
+    print(show_hidden_word(secret_word, old_letters_guessed))
+    return old_letters_guessed, num_of_tries
+
+def main():
+    opening_screen()
+    secret_word = choose_word(*get_data_from_user())
```

Comment:
no.

Reply:
Also why let the user chose what the word will be, just randomize it

Reply:
thats what i choose